docs/src/modules/ROOT/pages/integration/integration.adoc

```diff
@@ -271,22 +271,743 @@ To use Timefold Solver with Quarkus, read the xref:quickstart/quarkus/quarkus-qu
 If you are starting a new project, visit the https://code.quarkus.io/[code.quarkus.io] and select
 the _Timefold AI constraint solver_ extension before generating your application.
 
+=== Timefold configuration properties
+
 Following properties are supported in the Quarkus `application.properties`:
 
 :property_prefix: quarkus.
 include::config-properties.adoc[]
 
+[#integrationWithQuarkusManagedResources]
+=== Auto-managed resources
+
+The Quarkus integration allows the injection of several managed resources, including `SolverConfig`, `SolverFactory`,
+`SolverManager`, `SolutionManager`, `ScoreManager`, and `ConstraintVerifier`.
+
+The `SolverConfig` resource is constructed by reading the `application.properties` file and classpath. Therefore, Domain entities
+(solution, entities, and constraint classes) and customized properties (spent-limit, domain-access-type, etc) for the
+planning problem are identified and loaded into the solver configuration.
+
+The available resouses can be injected as follows:
+
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+@Path("/path")
+public class Resource {
+
+    @Inject
+    SolverConfig solverConfig;
+
+    @Inject
+    SolverFactory<Timetable> solverFactory;
+
+    @Inject
+    SolverManager<Timetable, String> solverManager;
+
+    @Inject
+    SolutionManager<Timetable, SimpleScore> simpleSolutionManager; // <1>
+
+    @Inject
+    ScoreManager<Timetable, SimpleScore> simpleScoreManager; // <1>
+
+    @Inject
+    ConstraintVerifier<TimetableConstraintProvider, Timetable> constraintVerifier;
+
+    ...
+}
+----
+<1> The following score types are available: `SimpleScore`, `SimpleLongScore`, `SimpleBigDecimalScore`, `HardSoftScore`, `HardSoftLongScore`, `HardSoftBigDecimalScore`, `HardMediumSoftScore`, `HardMediumSoftLongScore`, `HardMediumSoftBigDecimalScore`, `BendableScore`, `BendableLongScore`, and `BendableBigDecimalScore`.
```

Comment:
Don't we have a list of score types elsewhere in the docs? If so, I'd just reference that. No point in duplicating content.